### PR TITLE
Improve usage of `detect_from_dataframes` function

### LINKED ISSUE #2214: Improve usage of `detect_from_dataframes` function
### Problem Description
The metadata's `detect_from_dataframes` is currently meant to be used on a pre-existing instance of a metadata object. The function internally updates the metadata object and doesn't return anything.

The usage leads to some redundant code and confusion because you have to instantiate the object separately (why?), and it's unexpected that detection would not return anything. Additionally, we only expect this function to be used once, so it's not clear why it's possible to apply it to the same object multiple times.

### Expected behavior
To improve the usage of this function, we will make it a class function that ultimately returns an instance of the metadata object.

```python
from sdv.metadata import Metadata

metadata = Metadata.detect_from_dataframes(data=my_dataframes)
```

### Additional context
All detection functions will be updated to follow this paradigm. See:
- `detect_from_dataframe` (singular) improvements, specified in #2211 
- `detect_from_ddl` improvements specified in #2212

### PR DESCRIPTION
**Pull request by amontanez24**
resolve #2214 
CU-86b23k3j9

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b23k3j9 SDV - Improve usage of `detect_from_dataframes` function #2214](https://app.clickup.com/t/86b23k3j9)

**Review by lajohn4747 — APPROVED** [review]
LGTM

File: tests/unit/metadata/test_metadata.py [hunk 1]
```diff
@@ -539,6 +539,45 @@ def test_validate_data_no_relationships(self):
...
+        """
+        # Setup
+        mock_metadata.detect_table_from_dataframe = Mock()
+        mock_metadata._detect_relationships = Mock()
+        guests_table = pd.DataFrame()
+        hotels_table = pd.DataFrame()
+        data = {'guests': guests_table, 'hotels': hotels_table}
+
+        # Run
+        Metadata.detect_from_dataframes(data)
```
Comment (rwedge) [inline tests/unit/metadata/test_metadata.py:557]: should we also test the returned value?

**Review by rwedge — COMMENTED** [review]
Should any changes be made to `MultiTableMetadata.detect_from_dataframes`?

File: tests/unit/metadata/test_metadata.py (hunk 1 above)
Comment (amontanez24) [reply tests/unit/metadata/test_metadata.py:557]: yes, good catch!

**Review by rwedge — APPROVED** [review]

**Pull request merged**